spec.bs

```diff
@@ -6,57 +6,417 @@ Status: CG-DRAFT
+
+<xmp class='idl'>
+  [Exposed=(Window,SharedStorageWorklet)]
+  interface SharedStorage {
+    [Exposed=(Window)] 
+    Promise<any> run(DOMString name,
+                     optional SharedStorageRunOperationMethodOptions options = {});
+
+    [Exposed=(Window)] 
+    Promise<USVString> selectURL(DOMString name,
                                  FrozenArray<SharedStorageUrlWithMetadata> urls,
                                  optional SharedStorageRunOperationMethodOptions options = {});
 
-  Promise<any> run(DOMString name, optional SharedStorageRunOperationMethodOptions options = {});
+    Promise<any> set(DOMString key,
+                     DOMString value,
+                     optional SharedStorageSetMethodOptions options = {});
+
+    Promise<any> append(DOMString key,
+                        DOMString value);
+
+    Promise<any> delete(DOMString key);
+
+    Promise<any> clear();
+
+    [Exposed=(SharedStorageWorklet)] Promise<DOMString> get(DOMString key);
+
+    [Exposed=(SharedStorageWorklet)] Promise<unsigned long> length();
+
+    [Exposed=(SharedStorageWorklet)] Promise<double> remainingBudget();
 
-  readonly attribute SharedStorageWorklet worklet;
-};
+    [Exposed=(SharedStorageWorklet)] SharedStorageIterator keys();
 
-[Exposed=(Window)]
-interface SharedStorageWorklet {
-  Promise<undefined> addModule(USVString moduleURL);
-};
+    [Exposed=(SharedStorageWorklet)] SharedStorageIterator entries();
 
-dictionary SharedStorageSetMethodOptions {
-  boolean ignoreIfPresent;
-};
+    [Exposed=(Window)] readonly attribute SharedStorageWorklet worklet;
+  };
 
-dictionary SharedStorageRunOperationMethodOptions {
-  object data;
-};
+  dictionary SharedStorageSetMethodOptions {
+    boolean ignoreIfPresent;
+  };
 
-dictionary SharedStorageUrlWithMetadata {
-  required USVString url;
-  object reportingMetadata;
-};
+  dictionary SharedStorageRunOperationMethodOptions {
+    object data;
+  };
+
+  enum IteratorMode { "key", "keyValue" };
+
+  [Exposed=(SharedStorageWorklet)]
+  interface SharedStorageIterator {
+    constructor(IteratorMode mode);
+    Promise<object> next();
+  };
 </xmp>
+
+  ## Note on Parallelism ## {#parallelism}
+
+  The {{SharedStorage}} interface runs its methods <a href="https://html.spec.whatwg.org/multipage/infrastructure.html#in-parallel">in parallel</a>. To avoid race conditions within a single context, each context that calls into the interface will have its own <a href="https://html.spec.whatwg.org/multipage/infrastructure.html#parallel-queue">parallel queue</a> for handling operations.
+
+  In particular, the {{Window}}'s {{SharedStorage}} operations for a given frame will use a separate <a href="https://html.spec.whatwg.org/multipage/infrastructure.html#parallel-queue">parallel queue</a> from that used by any {{SharedStorageWorklet}} that may be spun up by this frame.
```

Comment:
For things that are linked multiple times, it's better to define in the <pre class="anchors"> section  and reference the terms via [=XXX=]. Example: https://github.com/patcg-individual-drafts/topics/blob/af873d0cfa07a174b5c3e980f23d9114b1188d07/spec.bs

Reply:
Done.